ipie/addons/thermal/trial/one_body.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2022 The ipie Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors: Fionn Malone <fmalone@google.com>
+#          Joonho Lee
+#
+
+import numpy
+import scipy.linalg
+
+from ipie.addons.thermal.estimators.particle_number import particle_number
+from ipie.addons.thermal.estimators.thermal import one_rdm_stable
+from ipie.addons.thermal.estimators.greens_function import greens_function
+from ipie.addons.thermal.trial.chem_pot import compute_rho, find_chemical_potential
+from ipie.utils.misc import update_stack
+
+
+class OneBody(object):
```

Comment:
The Class(object) syntax is deprecated in python (I know this is a hangover from legacy.)